tests/lib/utils.js

```diff
@@ -122,6 +122,31 @@ describe('utils', () => {
           );
         });
       });
+
+      for (const scopeOptions of [
+        { ignoreEval: true, ecmaVersion: 6, sourceType: 'script', nodejsScope: true },
+        { ignoreEval: true, ecmaVersion: 6, sourceType: 'script' },
+        { ignoreEval: true, ecmaVersion: 6, sourceType: 'module' },
+      ]) {
+        const ast = espree.parse(`
+          const create = () => {};
+          const meta = {};
+          module.exports = { create, meta };
+        `, { ecmaVersion: 6 });
+        const expected = {
+          create: { type: 'Identifier' },
+          meta: { type: 'Identifier' },
```

Comment:
This should still break rules like [this](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/02c1f347f2f0881e364724a4fce69e9da2ce9313/lib/rules/meta-property-ordering.js#L42), It should return the reference. But I'm not going to work on it in this PR.